src/Components/Blazor/Blazor/src/Hosting/WebAssemblyHostBuilder.cs

```diff
@@ -3,98 +3,100 @@
 using Microsoft.Extensions.DependencyInjection.Extensions;
 using Microsoft.Extensions.Logging;
 using Microsoft.JSInterop;
 
 namespace Microsoft.AspNetCore.Blazor.Hosting
 {
-    //
-    // This code was taken virtually as-is from the Microsoft.Extensions.Hosting project in aspnet/Hosting and then
-    // lots of things were removed.
-    //
-    internal class WebAssemblyHostBuilder : IWebAssemblyHostBuilder
+    /// <summary>
+    /// A builder for configuring and creating a <see cref="WebAssemblyHost"/>.
+    /// </summary>
+    public sealed class WebAssemblyHostBuilder
     {
-        private List<Action<WebAssemblyHostBuilderContext, IServiceCollection>> _configureServicesActions = new List<Action<WebAssemblyHostBuilderContext, IServiceCollection>>();
-        private bool _hostBuilt;
-        private WebAssemblyHostBuilderContext _BrowserHostBuilderContext;
-        private IWebAssemblyServiceFactoryAdapter _serviceProviderFactory = new WebAssemblyServiceFactoryAdapter<IServiceCollection>(new DefaultServiceProviderFactory());
-        private IServiceProvider _appServices;
-
         /// <summary>
-        /// A central location for sharing state between components during the host building process.
+        /// Creates an instance of <see cref="WebAssemblyHostBuilder"/> using the most common
+        /// conventions and settings.
         /// </summary>
-        public IDictionary<object, object> Properties { get; } = new Dictionary<object, object>();
+        /// <param name="args">The argument passed to the application's main method.</param>
+        /// <returns>A <see cref="WebAssemblyHostBuilder"/>.</returns>
+        public static WebAssemblyHostBuilder CreateDefault(string[] args = default)
+        {
+            // We don't use the args for anything right now, but we want to accept them
+            // here so that it shows up this way in the project templates.
+            args ??= Array.Empty<string>();
+            var builder = new WebAssemblyHostBuilder();
+
+            // Right now we don't have conventions or behaviors that are specific to this method
+            // however, making this the default for the template allows us to add things like that
+            // in the future, while giving `new WebAssemblyHostBuilder` as an opt-out of opinionated
+            // settings.
+            return builder;
+        }
 
         /// <summary>
-        /// Adds services to the container. This can be called multiple times and the results will be additive.
+        /// Creates an instance of <see cref="WebAssemblyHostBuilder"/> with the minimal configuration.
         /// </summary>
-        /// <param name="configureDelegate"></param>
-        /// <returns>The same instance of the <see cref="IWebAssemblyHostBuilder"/> for chaining.</returns>
-        public IWebAssemblyHostBuilder ConfigureServices(Action<WebAssemblyHostBuilderContext, IServiceCollection> configureDelegate)
+        public WebAssemblyHostBuilder()
```

Comment:
Until we reach a point where `CreateDefault` does something "opinionated", should we consider making this default constructor private? There's no use case for it today, so maybe we should leave open the option of not having it in the future.

Reply:
Sure, lets